Source/WebCore/loader/cache/CachedResourceLoader.cpp

```diff
@@ -1174,7 +1174,8 @@ ResourceErrorOr<CachedResourceHandle<CachedResource>> CachedResourceLoader::requ
             && !m_documentLoader->advancedPrivacyProtections().contains(AdvancedPrivacyProtections::HTTPSOnlyExplicitlyBypassedForDomain)
             && !frame->loader().isHTTPFallbackInProgress()
             && !(committedDocumentURL.protocolIs("http"_s) && request.resourceRequest().isSameSite());
-        if (!madeHTTPS && !request.resourceRequest().url().protocolIs("https"_s) && type == CachedResource::Type::MainResource && isHTTPSOnlyActive)
+        bool isAllowedSchemes = request.resourceRequest().url().protocolIs("https"_s) || request.resourceRequest().url().protocol().endsWith("-web-extension"_s);
```

Comment:
This is a layering violation.  WebKit should look in the place it is told about extension schemes instead.
Also, this would ideally have a test to verify it works end-to-end and to verify we won't break it again in the future.